### ISSUE TITLE
Bug: Problems with multiple scopes beginning with the same name

### ISSUE THREAD

**Issue by ghost** [binarylogic/searchlogic#11]
Please have a look at this small example:

 @@@ ruby
class Contact < ActiveRecord::Base
  has_many :documents, :as => :object
  has_many :involvements
  has_many :documents_involved, :through => :involvements, :source => :object, :source_type => &rsquo;Document&rsquo;
end

Contact.search(:documents_involved_title_like => &rsquo;foo&rsquo;).all
# => NoMethodError: undefined method `involved_title_like&rsquo; for #Class:...

 @@@

Using Rails 2.3.3 and searchlogic plugin 2.1.13 released 2009-07-29

[original LH ticket](http://binarylogic.lighthouseapp.com/projects/16601/tickets/111)

 This ticket has 0 attachment(s).

**Comment by ghost** [comment]
**Bug: Problems with multiple scopes beginning with the same name**

I am trying to create a failing test for this in my test suite and can&rsquo;t seem to do it. Do you mind giving that a shot. I am 90% certain this has to do with the order of keywords in the regular expression. Ex:

/(documents|documents_involved)/

When it should be

/(documents_involved|documents)/

This is a really easy fix, but I want to get a failing test in place first.

 by Ben Johnson

**Comment by ghost** [comment]
**Bug: Problems with multiple scopes beginning with the same name**

This has got a little bit more complicated because the bug has changed since searchlogic 2.3.3. Now it occurs not every time. I can&rsquo;t reproduce it and can&rsquo;t write a failing test, sorry. But it seems that it has something to do with other calls to Contact.search BEFORE - very strange.

But I have found a very similar issue, perhaps they are related. Here is a small patch for the specs (just add one line) and the specs will fail:

 @@@
diff --git a/vendor/plugins/searchlogic/spec/spec_helper.rb b/vendor/plugins/searchlogic/spec/spec_helper.rb
index a2b98f0..c155938 100644
--- a/vendor/plugins/searchlogic/spec/spec_helper.rb
+++ b/vendor/plugins/searchlogic/spec/spec_helper.rb
 @@ -71,6 +71,7 @@ Spec::Runner.configure do |config|
     class User < ActiveRecord::Base
       belongs_to :company, :counter_cache => true
       has_many :orders, :dependent => :destroy
- ```
   has_many :orders_big, :class_name => &rsquo;Order&rsquo;, :conditions => &rsquo;total > 100&rsquo;
  ```
  
   end
  
   class Order < ActiveRecord::Base
  @@@
  
  by Georg Ledermann

**Comment by ledermann** [comment]
Seems to be fixed by c8547d579312a2b7dbdfeac4ac964c3f7bb11972. Thanks!

**Comment by ghost** [comment]
Thanks
